substrate/client/basic-authorship/src/basic_authorship.rs

```diff
@@ -283,28 +283,48 @@ where
 		max_duration: time::Duration,
 		block_size_limit: Option<usize>,
 	) -> Self::Proposal {
-		let (tx, rx) = oneshot::channel();
-		let spawn_handle = self.spawn_handle.clone();
+		Self::propose(
+			self,
+			ProposeArgs {
+				inherent_data,
+				inherent_digests,
+				max_duration,
+				block_size_limit,
+				ignored_nodes_by_proof_recording: None,
+			},
+		)
+		.boxed()
+	}
+}
 
-		spawn_handle.spawn_blocking(
-			"basic-authorship-proposer",
-			None,
-			Box::pin(async move {
-				// leave some time for evaluation and block finalization (10%)
-				let deadline = (self.now)() + max_duration - max_duration / 10;
-				let res = self
-					.propose_with(inherent_data, inherent_digests, deadline, block_size_limit)
-					.await;
-				if tx.send(res).is_err() {
-					trace!(
-						target: LOG_TARGET,
-						"Could not send block production result to proposer!"
-					);
-				}
-			}),
-		);
+/// Arguments for [`Proposer::propose`].
+pub struct ProposeArgs<Block: BlockT> {
+	/// The inherent data to pass to the block production.
+	pub inherent_data: InherentData,
+	/// The inherent digests to include in the produced block.
+	pub inherent_digests: Digest,
+	/// Max duration for building the block.
+	pub max_duration: time::Duration,
+	/// Optional size limit for the produced block.
+	///
+	/// When set, block production ends before hitting this limit. The limit includes the storage
+	/// proof, when proof recording is activated.
+	pub block_size_limit: Option<usize>,
+	/// Trie nodes that should not be recorded.
+	///
+	/// Only applies when proof recording is enabled.
+	pub ignored_nodes_by_proof_recording: Option<IgnoredNodes<Block::Hash>>,
```

Comment:
Had a second thought here.

Maybe instead of providing some proof recording parameters here, it would be better to provide some callback which would allow caller of `Proposer` API customize the proof recorder instantiated by proposer internals? This could be more flexible for future extensions.

Something like:
```rust
		let maybe_proposal = self
			.proposer
			.propose(
				&parent_header,
				&inherent_data.0,
				inherent_data.1,
				Digest { logs: digest },
				proposal_duration,
				|proof_recorder| { proof_recorder.with_ignored_nodes(ignored_node) },
			)
			.await
			.map_err(|e| Box::new(e) as Box<dyn Error + Send>)?;

```